tests/forms_tests/tests/test_forms.py

```diff
@@ -3866,3 +3798,213 @@ class CustomForm(Form):
         custom = CustomRenderer()
         form = CustomForm(renderer=custom)
         self.assertEqual(form.renderer, custom)
+
+
+# RemovedInDjango50
+class DeprecationTests(SimpleTestCase):
+    def test_deprecation_warning_html_output(self):
+        msg = 'django.forms.BaseForm._html_output() is deprecated. Please use .render() and .get_context() instead.'
+        with self.assertRaisesMessage(RemovedInDjango50Warning, msg):
+            form = Person()
+            form._html_output(
+                normal_row='<p id="p_%(field_name)s"></p>',
+                error_row='%s',
+                row_ender='</p>',
+                help_text_html=' %s',
+                errors_on_separate_row=True,
+            )
+
+    def test_deprecation_warning_error_list(self):
+        class DivErrorList(ErrorList):
+            def __str__(self):
+                return self.as_divs()
+
+            def as_divs(self):
+                if not self:
+                    return ''
+                return '<div class="errorlist">%s</div>' % ''.join(
+                    '<div class="error">%s</div>' % e for e in self)
+
+        class EmailForm(Form):
+            email = EmailField()
+            comment = CharField()
+
+        data = {'email': 'invalid'}
+        f = EmailForm(data, error_class=DivErrorList)
+        msg = (
+            'Returning a plain string from ErrorList is deprecated. '
+            'Please customize via the template system instead.'
+        )
+        with self.assertRaisesMessage(RemovedInDjango50Warning, msg):
+            f.as_p()
+
+
+# RemovedInDjango50
+@ignore_warnings(category=RemovedInDjango50Warning)
+class DeprecatedTests(SimpleTestCase):
+    def test_errorlist_override_str(self):
+        class DivErrorList(ErrorList):
```

Comment:
Can we avoid duplicating this? Maybe define it at the module level as it is used in multiple test cases.